clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -329,7 +330,7 @@
                                         in("all", "-1", "0", "1"),
                                         Importance.LOW,
                                         ACKS_DOC)
-                                .define(COMPRESSION_TYPE_CONFIG, Type.STRING, "none", Importance.HIGH, COMPRESSION_TYPE_DOC)
+                                .define(COMPRESSION_TYPE_CONFIG, Type.STRING, CompressionType.NONE.name, in(CompressionType.names().toArray(new String[0])), Importance.HIGH, COMPRESSION_TYPE_DOC)
```

Comment:
Could we use [`Utils.enumOptions()`](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L1424) here?

Reply:
1. The parameters we need to pass in the validator here are `[none, gzip, snappy, lz4, zstd]`, not `[NONE, GZIP, SNAPPY, LZ4, ZSTD]`. So here I am not using the `Utils.enumOptions()` method directly.
2. Add the static method `names()` to `CompressionType`, so that this method can be called in places with the same requirements in the future.
WDYT?

Reply:
If we add a `toString()` method to `CompressionType`:
```
@Override
public String toString() {
    return name;
}
```

Then `Utils.enumOptions()` does what you want. Isn't that simpler?

Reply:
Well, I'll rework the code later. Thanks.